share/completions/mkvextract.fish

```diff
@@ -0,0 +1,47 @@
+end
+
+function __fish_print_mkvextract_tracks
+    mkvmerge -i $argv | string match 'Track ID*' | string replace -r '.*?(\d+): (.*)' '$1:\t$2'
+end
+
+# simple options
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2"    -s V -l "version" -d "Show version information"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2"    -s h -l "help"    -d "Show help"
+# commands
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2" -a "tracks"       -d "Extract tracks to external files"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2" -a "tags"         -d "Extract tags as XML"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2" -a "attachments"  -d "Extract attachments"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2" -a "chapters"     -d "Extract chapters as XML"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2" -a "cuesheet"     -d "Extract chapters and tags as CUE sheet"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2" -a "timecodes_v2" -d "Extract timecodes of a track as timecode v2 file"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -lt 2" -a "cues"         -d "Extract cue information as text file"
+# dynamic tracks and attachments completions
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3; and contains -- (commandline -opc)[2] tracks timecodes_v2 cues" -a "(__fish_print_mkvextract_tracks (commandline -opc)[3])"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3; and contains -- (commandline -opc)[2] attachments"              -a "(__fish_print_mkvextract_attachments (commandline -opc)[3])"
+# options common to all commands
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3"    -s f -l "parse-fully"          -d "Parse the whole file instead of relying on the index"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3"    -s v -l "verbose"              -d "Increase verbosity"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3"    -s q -l "quiet"                -d "Suppress status output"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3" -r      -l "ui-language"          -d "Force the translations for 'code' to be used"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3" -r      -l "command-line-charset" -d "Charset for strings on the command line"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3" -r      -l "output-charset"       -d "Outputs messages in specified charset"
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3" -r -s r -l "redirect-output"      -d "Redirect all messages into a file"
+# command-specific options
+complete -f -c mkvextract -n "test (count (commandline -opc)) -ge 3; and contains -- (commandline -opc)[2] tracks"   -r -s c                      -d "Convert text subtitles to a charset"
```

Comment:
We'd usually write this test as `__fish_seen_subcommand_from tracks` - that will also skip all options.